library/misc/datafixerupper/src/main/java/org/quiltmc/qsl/datafixerupper/impl/QuiltDataFixesInternals.java

```diff
@@ -16,100 +16,56 @@
 
 package org.quiltmc.qsl.datafixerupper.impl;
 
-import java.util.Collections;
-import java.util.Map;
-
-import com.mojang.datafixers.DataFixUtils;
 import com.mojang.datafixers.DataFixer;
 import com.mojang.datafixers.schemas.Schema;
-import com.mojang.logging.LogUtils;
-import com.mojang.serialization.Dynamic;
-import it.unimi.dsi.fastutil.objects.Object2ReferenceOpenHashMap;
 import org.jetbrains.annotations.*;
-import org.slf4j.Logger;
 
-import net.minecraft.SharedConstants;
 import net.minecraft.datafixer.DataFixTypes;
-import net.minecraft.datafixer.Schemas;
 import net.minecraft.nbt.NbtCompound;
-import net.minecraft.nbt.NbtOps;
+
+import org.quiltmc.loader.api.QuiltLoader;
 
 @ApiStatus.Internal
-public final class QuiltDataFixesInternals {
-	private QuiltDataFixesInternals() { }
+public abstract class QuiltDataFixesInternals {
+	public record DataFixerEntry(DataFixer dataFixer, int currentVersion) {}
 
-	public static final Logger LOGGER = LogUtils.getLogger();
+	@Contract(pure = true)
+	@Range(from = 0, to = Integer.MAX_VALUE)
+	public static int getModDataVersion(@NotNull NbtCompound compound, @NotNull String modId) {
+		return compound.getInt(modId + "_DataVersion");
+	}
 
-	private static Map<String, DataFixerEntry> modDataFixers = new Object2ReferenceOpenHashMap<>();
-	private static boolean frozen = false;
+	private static QuiltDataFixesInternals instance;
 
-	public static void registerFixer(@NotNull String modId,
-									 @Range(from = 0, to = Integer.MAX_VALUE) int currentVersion,
-									 @NotNull DataFixer dataFixer) {
-		if (modDataFixers.containsKey(modId)) {
-			throw new IllegalArgumentException("Mod '" + modId + "' already has a registered data fixer");
+	public static @NotNull QuiltDataFixesInternals get() {
+		if (instance == null) {
+			if (QuiltLoader.isModLoaded("databreaker")) {
```

Comment:
This doesn't feel like a good fix.
I'd rather avoid any modid hardcoding.

You said on Discord that with databreaker the Vanilla schemas would be null, I'd rather check that instead.

Reply:
Hm, that'd probably be better, yeah.